aws/resource_aws_glue_catalog_crawler.go

```diff
@@ -0,0 +1,411 @@
+	if len(targets) < 1 {
+		return []*glue.JdbcTarget{}
+	}
+
+	perms := make([]*glue.JdbcTarget, len(targets), len(targets))
+	for i, rawCfg := range targets {
+		cfg := rawCfg.(map[string]interface{})
+		perms[i] = expandJdbcTarget(cfg)
+	}
+	return perms
+}
+
+func expandJdbcTarget(cfg map[string]interface{}) *glue.JdbcTarget {
+	target := &glue.JdbcTarget{
+		Path:           aws.String(cfg["path"].(string)),
+		ConnectionName: aws.String(cfg["connection_name"].(string)),
+	}
+
+	if exclusions, ok := cfg["exclusions"]; ok {
+		target.Exclusions = expandStringList(exclusions.([]interface{}))
+	}
+	return target
+}
+
+func resourceAwsGlueCatalogCrawlerUpdate(d *schema.ResourceData, meta interface{}) error {
+	glueConn := meta.(*AWSClient).glueconn
+	name := d.Get("name").(string)
+
+	crawlerInput, err := createCrawlerInput(name, d)
+	if err != nil {
+		return err
+	}
+
+	crawlerUpdateInput := glue.UpdateCrawlerInput(*crawlerInput)
+	if _, err := glueConn.UpdateCrawler(&crawlerUpdateInput); err != nil {
+		return err
+	}
+
+	return resourceAwsGlueCatalogCrawlerRead(d, meta)
+}
+
+func resourceAwsGlueCatalogCrawlerRead(d *schema.ResourceData, meta interface{}) error {
+	glueConn := meta.(*AWSClient).glueconn
+	name := d.Get("name").(string)
+
+	input := &glue.GetCrawlerInput{
+		Name: aws.String(name),
+	}
+
+	crawlerOutput, err := glueConn.GetCrawler(input)
+	if err != nil {
+		if isAWSErr(err, glue.ErrCodeEntityNotFoundException, "") {
+			log.Printf("[WARN] Glue Crawler (%s) not found, removing from state", d.Id())
+			d.SetId("")
+		}
+
+		return fmt.Errorf("error reading Glue crawler: %s", err.Error())
+	}
+
+	d.Set("name", crawlerOutput.Crawler.Name)
+	d.Set("database_name", crawlerOutput.Crawler.DatabaseName)
+	d.Set("role", crawlerOutput.Crawler.Role)
+	d.Set("description", crawlerOutput.Crawler.Description)
+	d.Set("schedule", crawlerOutput.Crawler.Schedule)
+	d.Set("classifiers", flattenStringList(crawlerOutput.Crawler.Classifiers))
+	d.Set("table_prefix", crawlerOutput.Crawler.TablePrefix)
+
+	if crawlerOutput.Crawler.SchemaChangePolicy != nil {
+		schemaPolicy := map[string]string{
+			"delete_behavior": *crawlerOutput.Crawler.SchemaChangePolicy.DeleteBehavior,
+			"update_behavior": *crawlerOutput.Crawler.SchemaChangePolicy.UpdateBehavior,
+		}
+		d.Set("schema_change_policy", schemaPolicy)
+	}
+
+	var s3Targets = crawlerOutput.Crawler.Targets.S3Targets
+	if crawlerOutput.Crawler.Targets.S3Targets != nil {
+		if err := d.Set("s3_target", flattenS3Targets(s3Targets)); err != nil {
+			log.Printf("[ERR] Error setting Glue S3 Targets: %s", err)
+		}
+	}
+
+	var jdbcTargets = crawlerOutput.Crawler.Targets.JdbcTargets
+	if crawlerOutput.Crawler.Targets.JdbcTargets != nil {
+		if err := d.Set("jdbc_target", flattenJdbcTargets(jdbcTargets)); err != nil {
+			log.Printf("[ERR] Error setting Glue JDBC Targets: %s", err)
+		}
+	}
+
+	// AWS provides no other way to read back the additional_info
```

Comment:
The schema does not define anything relating to `additional_info`. Did you mean `configuration`? `crawlerOutput.Configuration` will have the information.